File: algorithms/python/two-sum.py
def twoSum(nums, target):
    """
    :param nums:
    :param target:
    :return:
    """
    indexDict = {}
    for key, val in enumerate(nums):
        another = target - val
        if another in indexDict:
            return [indexDict[another], key]
        indexDict[val] = key
    return None

File: algorithms/php/two-sum.php
<?php

    /**
     * @param int[] $nums
     * @param int $target
     * @return int[]
     */
    function twoSum($nums, $target)
    {
        $indexMap = [];

        foreach ($nums as $key => $val) {
            $another = $target - $val;
            if (array_key_exists($another, $indexMap)) return [$indexMap[$another], $key];
            $indexMap[$val] = $key;
        }
        return [];
    }